### PR TITLE
feat(rust, python): avoid panic error in strftime with invalid format

### PR DESCRIPTION
**Pull request by MarcoGorelli**
On the latest release it looks like this:
```python-traceback
In [32]: pl.Series(["2020-01-01"]).str.strptime(pl.Datetime).dt.strftime('%z')
thread '<unnamed>' panicked at 'a formatting trait implementation returned an error: Error', library/alloc/src/fmt.rs:612:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
Cell In [32], line 1
----> 1 pl.Series(["2020-01-01"]).str.strptime(pl.Datetime).dt.strftime('%z')

File ~/tmp/.venv/lib/python3.8/site-packages/polars/internals/series/utils.py:98, in call_expr.<locals>.wrapper(self, *args, **kwargs)
     96     expr = getattr(expr, namespace)
     97 f = getattr(expr, func.__name__)
---> 98 return s.to_frame().select(f(*args, **kwargs)).to_series()

File ~/tmp/.venv/lib/python3.8/site-packages/polars/internals/dataframe/frame.py:5732, in DataFrame.select(self, exprs, *more_exprs, **named_exprs)
   5616 def select(
   5617     self,
   5618     exprs: (
   (...)
   5627     **named_exprs: str | PolarsExprType | PythonLiteral | pli.Series | None,
   5628 ) -> Self:
   5629     """
   5630     Select columns from this DataFrame.
   5631
   (...)
   5729
   5730     """
   5731     return self._from_pydf(
-> 5732         self.lazy()
   5733         .select(exprs, *more_exprs, **named_exprs)
   5734         .collect(no_optimization=True)
   5735         ._df
   5736     )

File ~/tmp/.venv/lib/python3.8/site-packages/polars/internals/lazyframe/frame.py:1145, in LazyFrame.collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, no_optimization, slice_pushdown, common_subplan_elimination, streaming)
   1134     common_subplan_elimination = False
   1136 ldf = self._ldf.optimization_toggle(
   1137     type_coercion,
   1138     predicate_pushdown,
   (...)
   1143     streaming,
   1144 )
-> 1145 return pli.wrap_df(ldf.collect())

PanicException: a formatting trait implementation returned an error: Error
```

Here:
```python-traceback
In [1]: pl.Series(["2020-01-01"]).str.strptime(pl.Datetime).dt.strftime('%z')
thread '<unnamed>' panicked at 'a formatting trait implementation returned an error: Error', library/alloc/src/fmt.rs:612:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---------------------------------------------------------------------------
ComputeError                              Traceback (most recent call last)
Cell In[1], line 1
----> 1 pl.Series(["2020-01-01"]).str.strptime(pl.Datetime).dt.strftime('%z')

File ~/polars-dev/py-polars/polars/internals/series/utils.py:98, in call_expr.<locals>.wrapper(self, *args, **kwargs)
     96     expr = getattr(expr, namespace)
     97 f = getattr(expr, func.__name__)
---> 98 return s.to_frame().select(f(*args, **kwargs)).to_series()

File ~/polars-dev/py-polars/polars/internals/dataframe/frame.py:5766, in DataFrame.select(self, exprs, *more_exprs, **named_exprs)
   5650 def select(
   5651     self,
   5652     exprs: (
   (...)
   5661     **named_exprs: str | PolarsExprType | PythonLiteral | pli.Series | None,
   5662 ) -> Self:
   5663     """
   5664     Select columns from this DataFrame.
   5665 
   (...)
   5763 
   5764     """
   5765     return self._from_pydf(
-> 5766         self.lazy()
   5767         .select(exprs, *more_exprs, **named_exprs)
   5768         .collect(no_optimization=True)
   5769         ._df
   5770     )

File ~/polars-dev/py-polars/polars/internals/lazyframe/frame.py:1145, in LazyFrame.collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, no_optimization, slice_pushdown, common_subplan_elimination, streaming)
   1134     common_subplan_elimination = False
   1136 ldf = self._ldf.optimization_toggle(
   1137     type_coercion,
   1138     predicate_pushdown,
   (...)
   1143     streaming,
   1144 )
-> 1145 return pli.wrap_df(ldf.collect())

ComputeError: Cannot format NaiveDateTime with format '%z'.
```

### CODE REVIEW

File: polars/polars-core/src/chunked_array/temporal/datetime.rs [hunk 1]
```diff
@@ -207,14 +207,31 @@ impl DatetimeChunked {
...
+        match self.time_zone() {
             #[cfg(feature = "timezones")]
-            Some(_) => format!(
+            Some(_) => match write!(
+                fmted,
                 "{}",
                 Utc.from_local_datetime(&dt).earliest().unwrap().format(fmt)
-            ),
-            _ => format!("{}", dt.format(fmt)),
+            ) {
```
Comment (ritchie46) [inline polars/polars-core/src/chunked_array/temporal/datetime.rs:217]: I think it is cleaner to write something like this:

```rust
write!().map_err(|_| PolarsError::ComputeError(..))?;
```

File: polars/polars-core/src/chunked_array/temporal/datetime.rs [hunk 2]
```diff
@@ -207,14 +207,31 @@ impl DatetimeChunked {
...
+                Ok(_) => (),
+                Err(_) => {
+                    return Err(PolarsError::ComputeError(
+                        format!("Cannot format DateTime with format '{fmt}'.").into(),
+                    ));
+                }
+            },
+            _ => match write!(fmted, "{}", dt.format(fmt)) {
```
Comment (ritchie46) [inline polars/polars-core/src/chunked_array/temporal/datetime.rs:225]: See comment above.

**Review by ritchie46 — COMMENTED** [review]
Thanks @MarcoGorelli. I have a small clippy like comment. Other than than that great additon. Errors > panics!

File: polars/polars-core/src/chunked_array/temporal/datetime.rs (hunk 1 above)
Comment (MarcoGorelli) [reply polars/polars-core/src/chunked_array/temporal/datetime.rs]: right, that's better - thanks!

**Pull request merged**